packages/terra-date-time-picker/src/DateTimeUtils.js

```diff
@@ -75,37 +75,58 @@ class DateTimeUtils {
    * Synchronize only the time for a given moment object.
    * @param {object} momentDate - The moment object to synchronize the date and time.
    * @param {string} time - The time to synchronize with the moment object.
+   * @param {boolean} hasSeconds - If true seconds will be processed
    * @return {object} - The synchronized moment object.
    */
-  static updateTime(momentDate, time) {
+  static updateTime(momentDate, time, hasSeconds) {
```

Comment:
There are a few places [[1](https://github.com/cerner/terra-framework/blob/master/packages/terra-date-time-picker/src/DateTimeUtils.js#L52-L54)] [[2](https://github.com/cerner/terra-framework/blob/master/packages/terra-date-time-picker/src/DateTimeUtils.js#L63-L65)] [[3](https://github.com/cerner/terra-framework/blob/master/packages/terra-date-time-picker/src/DateTimeUtils.js#L173)] in DateTimeUtils that calls `updateTime`. Those need to pass down the hasSeconds parameter and need to check if the length of the time string is 5 or 8 depending if seconds is included.

Reply:
Fixed in 873a928b83c481df3173645df2680d0e54f16fa0